### ISSUE TITLE
False positive: possibly-used-before-assignment and sys.exit()

### ISSUE THREAD

**Issue by mueller-ma** [pylint-dev/pylint#9664]
### Bug description

```python
import sys

if True:
    x = 0
else:
    sys.exit(0)

print(x) # Reports possibly-used-before-assignment

```

### Command used

```shell
pylint foo.py
```

### Pylint output

```python
************* Module foo
[Some unrelated messages]
foo.py:8:6: E0606: Possibly using variable 'x' before assignment (possibly-used-before-assignment)
```

### Expected behavior

No error

### Pylint version

```shell
pylint 3.2.0
astroid 3.2.2
Python 3.11.2 (main, Mar 13 2023, 12:18:29) [GCC 12.2.0]
```

**Comment by mueller-ma** [comment]
#9627

**Comment by Pierre-Sassoulas** [comment]
Thank you for the triaging @mueller-ma 😄
